crates/provider/src/heart.rs

```diff
@@ -353,6 +354,9 @@ pub enum WatchTxError {
     /// Transaction was not confirmed after configured timeout.
     #[error("transaction was not confirmed within the timeout")]
     Timeout,
+    /// Transaction was dropped from the mempool.
+    #[error("transaction was dropped from the mempool")]
+    Dropped,
```

Comment:
This change is technically breaking, as the enum is not marked with `#[non_exhaustive]`, but I'm not sure what's the project policy in that case is.